### ISSUE TITLE
Minimal trainer for small models (150–350M params)

### ISSUE THREAD

**Issue by z-a-f** [maida-ai/Research#7]
Train tiny GPT-style models using DP-ASSM or BLADE blocks.

## Description

Provide a thin, readable training loop (or HF `Trainer`) with logging and checkpoints, so we can validate learning and run ablations.

## Implementation Suggestions

* File: `efficient_longctx/training/train.py`
* Create `LongCtxModel`:
  * Token embedding → N layers of chosen block (`--block dpassm|blade|vanilla`) → LM head.
* Add AMP/bfloat16, gradient accumulation, gradient clipping, TensorBoard logging.
* Example:
  ```
  python -m efficient_longctx.training.train \
    --dataset openwebtext --seq_len 2048 --params 150m \
    --block dpassm --window_size 2048 --ssm_state_dim 256 \
    --batch_size 16 --lr 3e-4 --epochs 1 --out_dir runs/dpassm_150m/
  ```

## Acceptance Criteria

* [ ] Training runs to completion on a single GPU and saves checkpoints.
* [ ] Loss decreases on synthetic + tiny real corpus splits.
* [ ] TensorBoard shows loss/throughput logs.
